crates/router/src/core/cards_info.rs

```diff
@@ -0,0 +1,26 @@
+use error_stack::{report, ResultExt};
+use router_env::{instrument, tracing};
+
+use crate::{
+    core::errors::{self, RouterResponse},
+    db::StorageInterface,
+    services::ApplicationResponse,
+    types::transformers::ForeignFrom,
+};
+
+#[instrument(skip_all)]
+pub async fn retrieve_card_info(
+    store: &dyn StorageInterface,
+    card_iin: String,
+) -> RouterResponse<api_models::cards_info::CardInfoResponse> {
```

Comment:
Shouldn't we validate that the IIN/BIN is a min. of 6 and a max. of 9 characters only?